mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -70,12 +70,148 @@ macro_rules! event {
     }
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+pub enum AddressingStyle {
+    /// Use virtual addressing if possible, but fall back to path addressing if necessary
+    #[default]
+    Automatic,
+    /// Always use virtual addressing
+    Virtual,
+    /// Always use path addressing
+    Path,
+}
+
+#[derive(Debug, Clone)]
+pub struct EndpointConfig {
+    region: String,
+    use_fips: bool,
+    use_accelerate: bool,
+    use_dual_stack: bool,
+    endpoint: Option<Uri>,
+    addressing_style: AddressingStyle,
+}
```

Comment:
Having the struct out to a file `endpoint_config.rs` would require either making its fields public or having getter methods for its field, which we wanted to avoid right?
I can add the tests here.

Reply:
As discussed, better to move into a module and accept creating the getter methods. Makes the code easier to understand and test in isolation.

Getters aren't always bad - just when they don't provide any benefit within the same module.

(I see this was updated in the last commit, resolving)